Add Latin Hypercube Sampling for Monte Carlo Analysis (#57)
ReEDS-Model/ReEDS: docs/source/user_guide.md

3. Set `MCS_dist_groups` to one or more YAML group names. Separate multiple groups with a dot.
   For example `tech.load_st.ng_fuel_price`.

4. Choose the sampling method; `MCS_lhs=1` uses a Latin Hypercube sampling method and `MCS_lhs=0` uses random sampling.

5. Run ReEDS as usual. Each Monte Carlo draw will create its own run using the sampled inputs.

> These four switches (`MCS_runs`,`MCS_dist`, `MCS_dist_groups`, and `MCS_lhs`) are the only required controls.

Reviewer: ```suggestion
These four switches (`MCS_runs`, `MCS_dist`, `MCS_dist_groups`, and `MCS_lhs`) are the only required controls.
```

This enables state level uncertainty in siting supply curves for wind and solar technologies through a random draw between `limited` and `reference` conditions.

### Sampling method

There are two sampling approaches available for Monte Carlo analysis: random sampling and Latin Hypercube sampling. 

> Random sampling uses the numpy `random` method for the relevant distribtion to sample a set of weights. 

Reviewer: ```suggestion
Random sampling uses the numpy `random` method for the relevant distribution to sample a set of weights. 
```

These weights are applied to the assignment values specified in the distribution group to generate the value for each run.
To ensure reproducibility, the Monte Carlo run number is used as the seed value. 

The second approach, Latin Hypercube sampling (LHS), utilizes a quasi-Monte Carlo sampling method that is 
designed to improve efficiency by reducing overlap of the sampled values. An overview of this method
can be found in {cite}`sheikholeslamiProgressiveLatinHypercube2017`. For this method, an NxD matrix is generated upfront
for all model runs based on the number of samples (N) and the independent dimensions being sampled (D). 
The values in this matrix represent sampling of the cumulative probability distribution functions, 
and are later used by the inverse CDF (percent point) functions to derive the actual sample values. 
A single seed value is used for all runs, resulting in unique sampling matrices for a given set of values of N and D.

> The LHS method tends to results in sampling value that converge on the true input distributions for lower numbers of samples.

Reviewer: ```suggestion
The LHS method tends to result in sampling values that converge on the true input distributions for lower numbers of samples.
```

Since these weights are used as multiplicative factors on reference files,
Dirichlet sampling tends to favor combinations that lean toward central scenarios when the inputs represent "low",
"medium", and "high" pathways.

> - 

Reviewer 2: ```suggestion
```
